packages/vsx-registry/src/browser/vsx-extensions-model.ts

```diff
@@ -139,12 +139,17 @@ export class VSXExtensionsModel {
             const searchResult = new Set<string>();
             for (const data of result.extensions) {
                 const id = data.namespace.toLowerCase() + '.' + data.name.toLowerCase();
+                const extension = this.api.getLatestCompatibleVersion(data.allVersions);
+                if (!extension) {
+                    continue;
+                }
                 this.setExtension(id).update(Object.assign(data, {
                     publisher: data.namespace,
-                    downloadUrl: data.files.download,
+                    downloadUrl: extension.url,
```

Comment:
Are you sure about this change? The `url` field in the extension metadata refers to the metadata API.